internal/helm/helm.go

```diff
@@ -80,7 +82,9 @@ func DiscoverChartVersions(
 
 	// NB: semver.Collection sorts in ascending order by default. We want to
 	// return the versions in descending order.
-	sort.Sort(sort.Reverse(semvers))
+	slices.SortFunc(semvers, func(a, b *semver.Version) int {
+		return b.Compare(a)
```

Comment:
The `semver` package has some issues with how it does comparisons. It treats `1.0` and `1.0.0`, for instance, as equal, which may be semantically true, _but_ it leads to sorts that might sometimes put `1.0` ahead of `1.0.0` while other times putting `1.0.0` ahead of `1.0`.

This ends up being the right way to handle:

```suggestion
		if comp := b.Compare(a); comp != 0 {
			return comp
		}
		return strings.Compare(b.Original(), a.Original())
```